frame/sgx-hello-world/src/lib.rs

```diff
@@ -169,7 +169,8 @@ decl_event!(
 	pub enum Event<T> where AccountId = <T as frame_system::Trait>::AccountId {
 		EnclaveAdded(AccountId),
 		EnclaveRemoved(AccountId),
-		EnclaveCallDispatched(AccountId),
+		EnclaveCallSuccess(AccountId),
+		EnclaveCallFailure(AccountId),
```

Comment:
Would be cool we could have a hash(account_id, call) then the `fake-mixer` could deduce which call got dispatched or failed.

Assuming that `call` is unique in this context.

The macro seems not to work for tuples tried `((AccountId, Vec<u8>))`

Reply:
> hash(account_id, call) 

Would extrinsic hash work? If not, why not?

Reply:
Not sure if it's a good idea or not, but would this work?

```
pub enum CallOutcome {
  WentSmoooooth,
  Naaah,
}
```
…and then do:
```
…
EnclaveCallDispatched(CallOutcome),
…
```

Reply:
> Would extrinsic hash work? If not, why not?

Well, the extrinsic of the executed call in the enclave is not returned back to the `submit-extrinsic` and it depends on some stuff that changes during runtime such as nonce that we can not know about in advance. Even though the orginal call is constructed and known in `submit-extrinsic`. However, the encrypted payload will be intact and should be possible to subscribe to.

Thus, I implemented `T::Hasher::hash_of(&(&account_id, &encrypted_payload))` which should be possible to construct in the same way in `submit-extrinsic`

Ok?